### PR TITLE
[HIP][COMGR] Re-enable PCH starting from 5.0 (SWDEV-308265 is fixed in ROCClr)

### PR DESCRIPTION
**Pull request by atamazov**
This switches off the W/A for [SWDEV-308265 "[COMGR] Precompiler headers are broken: __hipGetPCH() is not available.."](https://ontrack-internal.amd.com/browse/SWDEV-308265) introduced in #1243 starting from HIP version 5.0.0.

:warning: The fix in ROCClr present in Mainline since build 9292. However, the build reports HIP version 4.4.21493. In order to test PCH functionality in the Mainline, it is necessary to provide two additional options to MIOpen's CMake command line:
```
... -DMIOPEN_OVERRIDE_HIP_VERSION_MAJOR=5 -DMIOPEN_OVERRIDE_HIP_VERSION_MINOR=0 ...
```
Locally tested.

:warning: Similar PR needs to be prepared for 4.5.x (if 4.5.x with fix for missing `__hipGetPCH()` will be released).

### CODE REVIEW

**Comment by junliume** [comment]
@sunway513 should we cherry-pick it into 4.5.x as @atamazov  has mentioned above?

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@junliume This will only be required if a decision is made to release 4.5.3. Or if a decision is made to release PyTorch Wheels based on 4.5.2 with a fixed ROCClr. Either of these decisions require participation/approval of release managers, I guess.

File: src/comgr.cpp
Comment (atamazov) [inline src/comgr.cpp:111]: @junliume 
> @sunway513 should we cherry-pick it into 4.5.x as @atamazov has mentioned above?

Simple cherry-picking won't work. We'll need to also change 
```
(HIP_PACKAGE_VERSION_FLAT < 5000000000)
```
to
```
(HIP_PACKAGE_VERSION_FLAT < 4005<patch>)
```
where `<patch>` should be taken from HIP version of 4.5.x that contains fixed ROCClr.

**Comment by sunway513** [comment]
Let's move the discussion to internal chats, thanks.

**Pull request merged**
